go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -311,6 +361,287 @@ func registerUser(userAddress string, event []UserEventType, subsIdStr string) {
 	userSubscriptionMap[subsId] = userAddress
 }
 
+func checkNotificationDistancePeriodicTrigger() {
+
+	//only check if there is at least one subscription
+	mutex.Lock()
+	defer mutex.Unlock()
+	//check all that applies
+	for subsId, distanceCheck := range distanceSubscriptionMap {
+		if distanceCheck != nil && distanceCheck.Subscription != nil {
+			//decrement the next time to send a message
+			distanceCheck.NextTts--
+			if distanceCheck.NextTts > 0 {
+				continue
+			} else { //restart the nextTts and continue processing to send notification or not
+				distanceCheck.NextTts = distanceCheck.Subscription.Frequency
+			}
+			//loop through every reference address
+			returnAddr := make(map[string]*gisClient.Distance)
+			skipThisSubscription := false
+
+			//if reference address is specified, reference addresses are checked agains each monitored address
+			//if reference address is nil, each pair of the monitored address should be checked
+			//creating address pairs to check
+			//e.g. refAddr = A, B ; monitoredAddr = C, D, E ; resultingPairs {A,C - A,D - A,E - B,C - B,D - B-E}
+			//e.g. monitoredAddr = A, B, C ; resultingPairs {A,B - B,A - A,C - C,A - B,C - C,B}
+
+			var addressPairs []Pair
+			if distanceCheck.Subscription.ReferenceAddress != nil {
+				for _, refAddr := range distanceCheck.Subscription.ReferenceAddress {
+					//loop through every monitored address
+					for _, monitoredAddr := range distanceCheck.Subscription.MonitoredAddress {
+						pair := Pair{addr1: refAddr, addr2: monitoredAddr}
+						addressPairs = append(addressPairs, pair)
+					}
+				}
+			} else {
+				nbIndex := len(distanceCheck.Subscription.MonitoredAddress)
+				for i := 0; i < nbIndex-1; i++ {
+					for j := i + 1; j < nbIndex; j++ {
+						pair := Pair{addr1: distanceCheck.Subscription.MonitoredAddress[i], addr2: distanceCheck.Subscription.MonitoredAddress[j]}
+						addressPairs = append(addressPairs, pair)
+						//need pair to be symmetrical so that each is used as reference point and monitored address
+						pair = Pair{addr1: distanceCheck.Subscription.MonitoredAddress[j], addr2: distanceCheck.Subscription.MonitoredAddress[i]}
+						addressPairs = append(addressPairs, pair)
+					}
+				}
+			}
+
+			for _, pair := range addressPairs {
+				refAddr := pair.addr1
+				monitoredAddr := pair.addr2
+
+				var distParam gisClient.TargetPoint
+				distParam.AssetName = monitoredAddr
+
+				distResp, _, err := gisAppClient.GeospatialDataApi.GetDistanceGeoDataByName(context.TODO(), refAddr, distParam)
+				if err != nil {
+					log.Error("Failed to communicate with gis engine: ", err)
+					return
+				}
+
+				distance := int32(distResp.Distance)
+
+				switch *distanceCheck.Subscription.Criteria {
+				case ALL_WITHIN_DISTANCE:
+					if float32(distance) < distanceCheck.Subscription.Distance {
+						returnAddr[monitoredAddr] = &distResp
+					} else {
+						skipThisSubscription = true
+						break
```

Comment:
You are breaking from the switch statement only and not the addrPairs loop. You skip the subscription, however you may also want to stop sending GIS Engine REST API calls for any remaining address pairs for the current subscription.

Reply:
intention was to break from the addrPair loop as you are pointing out not only the innerloop.. since no break is needed in golang, i assumed it would break from the surrounding for loop.. but after double checking your comment is right on that matter. 
code updated